### ISSUE TITLE
Performance of Alpine vs. Ubuntu based docker images of Clickhouse: any difference?

### ISSUE THREAD

**Issue by mkaze** [ClickHouse/ClickHouse#63658]
In general case, the Alpine based images are favored mostly because of smaller image size, but in the past few years there were also some concerns/discussions about their slower performance for some cases/applications (e.g. [this](https://news.ycombinator.com/item?id=19861921), [this](https://superuser.com/q/1219609/473340), [this](https://www.reddit.com/r/node/comments/vdkf3a/to_alpine_or_not_to_alpine/)). 

Now, I was wondering if there is any difference between using the Alpine vs. Ubuntu based docker image of ClickHouse **in terms of performance**, or is there any specific recommendations/extra configurations to be aware of when using Alpine based images of ClickHouse in production? (I'll also add that we are using ClickHouse in a single-node self-managed deployment setup).

**Comment by 0xqd** [comment]
Alpine with gcc tends to have equivalent performance with ubuntu. Alpine with musl has worse performance most of the the time for all apps including Clickhouse.

**Comment by alexey-milovidov** [comment]
There is no difference in performance between these two images at all.

**Comment by alexey-milovidov** [comment]
ClickHouse on Alpine uses glibc, similar to the Ubuntu image.

**Comment by alexey-milovidov** [comment]
@0xqd, sorry, your comment has a mistake - you confused gcc and glibc.

**Comment by hoxnox** [comment]
I experienced performance issues during insert with clickhouse client from sql file on alpine images.

tried
```
clickhouse/clickhouse-server   23.9.5-alpine
clickhouse/clickhouse-server   23.8.8.20-alpine
```

I have file with ~5 million rows file. With `insert into ... values(),(),()` bulks of 1000.
Insert command:
```sh
docker exec -it clickhouse1.hbfw.ru clickhouse client --user user --pass pass --database db -nm --queries-file=/tmp/clickhouse.sql
```

Very poor performance ~1000 records per second.

With default latest image everything is fine - whole file load takes a few seconds.
```
clickhouse/clickhouse-server   latest
```

**Comment by hoxnox** [comment]
It looks like [my problem](https://github.com/ClickHouse/ClickHouse/issues/63658#issuecomment-2155852592) is not connected to alpine. With image `clickhouse/clickhouse-server   24.4.2.141` i have the same performance problems. Maybe something was fixed in lates release (24.5.1.1763).
